source/Cosmos.System2/FileSystem/Disk.cs

```diff
@@ -0,0 +1,296 @@
+﻿//#define COSMOSDEBUG
+
+using System;
+using System.Collections.Generic;
+using Cosmos.HAL.BlockDevice;
+using Cosmos.System.FileSystem.FAT;
+using Cosmos.System.FileSystem.VFS;
+
+namespace Cosmos.System.FileSystem
+{
+    public class Disk
+    {
+        private List<ManagedPartition> parts = new List<ManagedPartition>();
+        private static List<FileSystemFactory> registeredFileSystems = new List<FileSystemFactory>();
+        public bool IsMBR { get { return !GPT.IsGPTPartition(Host); } }
+        /// <summary>
+        /// The size of the disk in MB.
+        /// </summary>
+        public int Size { get; }
+        /// <summary>
+        /// List of partitions
+        /// </summary>
+        public List<ManagedPartition> Partitions
+        {
+            get
+            {
+                var converted = new List<ManagedPartition>();
+                if (GPT.IsGPTPartition(Host))
+                {
+                    var gpt = new GPT(Host);
+                    int i = 0;
+                    foreach (var item in gpt.Partitions)
+                    {
+                        var part = new ManagedPartition(new Partition(Host, item.StartSector, item.SectorCount));
+                        if (MountedPartitions[i] != null)
+                        {
+                            var data = MountedPartitions[i];
+                            part.RootPath = data.RootPath;
+                            part.MountedFS = data;
+                        }
+                        converted.Add(part);
+                        i++;
+                    }
+                }
+                else
+                {
+                    var mbr = new MBR(Host);
+                    int i = 0;
+                    foreach (var item in mbr.Partitions)
+                    {
+                        var part = new ManagedPartition(new Partition(Host, item.StartSector, item.SectorCount));
+                        if (MountedPartitions[i] != null)
+                        {
+                            var data = MountedPartitions[i];
+                            part.RootPath = data.RootPath;
+                            part.MountedFS = data;
+                        }
+                        converted.Add(part);
+                        i++;
+                    }
+                }
+
+                return converted;
+            }
+        }
+        /// <summary>
+        /// List of file systems.
+        /// </summary>
+        public static List<FileSystemFactory> RegisteredFileSystemsTypes { get { return registeredFileSystems; } }
+        /// <summary>
+        /// Main blockdevice that has all of the partitions.
+        /// </summary>
+        public BlockDevice Host;
+        public BlockDeviceType Type { get { return Host.Type; } }
+        public Disk(BlockDevice mainBlockDevice)
+        {
+            Host = mainBlockDevice;
+            if (registeredFileSystems.Count == 0)
```

Comment:
CAn we not set a default value for the List including the FatFileSystemFactory already?